### ISSUE TITLE
It appears there are two competing dhcp clients in Garden Linux 9xx

### ISSUE THREAD

**Issue by marwinski** [gardenlinux/gardenlinux#1319]
**What happened:**

I can find code and logs for two dhcp clients obtaining information via dhcp. Besides systemd-networkd there is a second package 
`isc-dhcp-client` which has apparently been installed by cloud-init:

```
root@vm-gl-test-20221005151209:/home/azureuser# apt-cache rdepends isc-dhcp-client
isc-dhcp-client
Reverse Depends:
  cloud-init
 |netscript-2.4
  ifupdown2
 |ifupdown-ng
 |ifupdown
  dracut-network
```

**Debug Logs:**

Log from systemd-networkd:

```
Oct 05 15:16:39 vm-gl-test-20221005151209 systemd-networkd[482]: eth0: DHCPv4 address 10.0.0.4/24, gateway 10.0.0.1 acquired from 168.63.129.16
```

Log from isc-dhcp-client:

```
Oct 05 15:16:38 localhost dhclient[464]: Internet Systems Consortium DHCP Client 4.4.3
Oct 05 15:16:38 localhost dhclient[464]: Copyright 2004-2022 Internet Systems Consortium.
Oct 05 15:16:38 localhost dhclient[464]: All rights reserved.
Oct 05 15:16:38 localhost dhclient[464]: For info, please visit https://www.isc.org/software/dhcp/
Oct 05 15:16:38 localhost dhclient[464]:
Oct 05 15:16:38 localhost dhclient[464]: Listening on LPF/eth0/60:45:bd:8e:42:ce
Oct 05 15:16:38 localhost dhclient[464]: Sending on   LPF/eth0/60:45:bd:8e:42:ce
Oct 05 15:16:38 localhost dhclient[464]: Sending on   Socket/fallback
Oct 05 15:16:38 localhost dhclient[464]: DHCPDISCOVER on eth0 to 255.255.255.255 port 67 interval 8
Oct 05 15:16:38 localhost dhclient[464]: DHCPOFFER of 10.0.0.4 from 168.63.129.16
Oct 05 15:16:38 localhost dhclient[464]: DHCPREQUEST for 10.0.0.4 on eth0 to 255.255.255.255 port 67
Oct 05 15:16:38 localhost dhclient[464]: DHCPACK of 10.0.0.4 from 168.63.129.16
Oct 05 15:16:38 localhost dhclient[464]: Timeout too large reducing to: 2147483646 (TIME_MAX - 1)
Oct 05 15:16:38 localhost dhclient[464]: bound to 10.0.0.4 -- renewal in 4294967295 seconds.
```

**How to reproduce it**:

I suppose any system with cloud-init. From what I can see is that network config is disabled (on azure) but this does not appear to work.

**Comment by marwinski** [comment]
An now we even have security vulnerabilities in the isc-dhcp-client :-( 
https://security-tracker.debian.org/tracker/source-package/isc-dhcp
We have version 4.4.3-2

**Comment by gyptazy** [comment]
Hey @marwinski,

seems like we could easily remove the dependency on `isc-dhcp-client` in `cloud-init` and re-package this within our GitLab pipeline. Within my local test build I had no further issues so we might give it a try.

Just let me know if this is fine for you and I should proceed. If yes, just assign this task on me.

CC: @nanory @waldiTM

**Comment by marwinski** [comment]
@gyptazy please go ahead

**Comment by gyptazy** [comment]
Hey @marwinski,

as discussed within the last weekly this might be our approach. Have you already validated this package on our primary hyperscalers? If everything is fine I may proceed with releasing this as a Garden Linux package on our mirrors.

**Comment by Vincinator** [comment]
Discussed with @marwinski and @gehoern, please go ahead with the release :)

**Comment by gyptazy** [comment]
> Discussed with @marwinski and @gehoern, please go ahead with the release :)

As I can see you were faster :) https://gitlab.com/gardenlinux/gardenlinux-package-build/-/merge_requests/155 technically approved (but should be tested as local_package before). Merge depends on you. Thanks!

**Comment by marwinski** [comment]
The fix does not appear to work, at least on Azure. When applied 

- the ssh key is not written to `/root/.ssh/authorized_key
- The VM does not reach ready state on azure

The reason for that is not obvious. cloud-init does not obviously print any error messages but appears to run just fine.

**Comment by marwinski** [comment]
It appears that cloud-init-local is started before the network, however Azure datasource requests network which fails to be initialised since we removed the dhcp client. Without network however, cloud-init cannot retrieve the public key for the azureuser at that time.

**Comment by gyptazy** [comment]
> It appears that cloud-init-local is started before the network, however Azure datasource requests network which fails to be initialised since we removed the dhcp client. Without network however, cloud-init cannot retrieve the public key for the azureuser at that time.

@marwinski that's a good hint, maybe we can just overwrite the systemd-unit to start after the network which is handled by Garden Linux's default dhcp client.

**Comment by marwinski** [comment]
yes, cloud-init-local should run after network initialisation, however this might have unforeseen side effects ...

**Comment by waldiTM** [comment]
Well. `cloud-init-local` is to be run before network, so it can configure network. The fallback to run after the system network setup is not supported anymore.

**Comment by marwinski** [comment]
so, then we should probably revert the patch and make sure we get an isc-dhcp-client that has no CVE.

**Comment by gyptazy** [comment]
**Update**:
 * Validate if there’s really no option to gather extended DHCP options by systemd-dhcp
 * If not, we'll mock this and provide additional unit test to ensure the mocks are still up to date
